src/site_upload/process_upload/process_upload.py

```diff
@@ -23,15 +23,35 @@ def process_upload(s3_client, sns_client, s3_bucket_name: str, s3_key: str) -> N
     metadata = functions.read_metadata(s3_client, s3_bucket_name)
     path_params = s3_key.split("/")
     study = path_params[1]
-    data_package = path_params[2]
+    # This happens when we're processing flat files, due to having to condense one
+    # folder layer.
+    # TODO: revisit on targeted crawling
+    if "__" in path_params[2]:
+        data_package = path_params[2].split("__")[1]
+    else:
+        data_package = path_params[2]
```

Comment:
nit: I feel like this repo is due for a AggPath class or similar that abstracts all of the path assumptions / evolutions over time. There's a lot of path[2] and splitting going on across the repo, instead of in one nice central place.

Reply:
yeah, that's really good idea. #143